### ISSUE TITLE
Location uses incorrect ToString method when generating LocationString

### ISSUE THREAD

**Issue by jwkerley** [maximn/google-maps#70]
In GoogleMapsApi.Entities.Common the LocationString property uses num.ToString(CultureInfo.InvariantCulture) for the Latitude and Longitude.

Sometimes double gives scientific notation, e.g. if I have a longitude of 0.000009 this is converted to "9E-06". Google maps API does not support this notation for lat/long.

According to StackOverflow the only way to reliable generate string in the correct format is to use num.ToString("0." + new string('#', 339)).TrimEnd('0');

This works for me if I create a new MyLocation class inherited from Location and override ToString to use this format.

```
  private class MyLocation : Location
        {
            public MyLocation(double lat, double lng) : base(lat, lng)
            {
            }

            public override string ToString()
            {
                string lats = ToNonScientificString(Latitude);
                string lons = ToNonScientificString(Longitude);
                return lats + "," + lons;
            }

            private static string ToNonScientificString(double d)
            {
                return d.ToString(DoubleFormat).TrimEnd('0');
            }

            private static readonly string DoubleFormat = "0." + new string('#', 339);
        }
```

**Comment by maximn** [comment]
Cool! I encourage you to issue a pull request with that fix :)

**Comment by alowdon** [comment]
Not sure if it's related to this change, but I've noticed that if you specify 0 for longitude then the request will fail because the generated location string has the longitude missing, e.g. if you tried to do a request for (52.123123, 0.0) then the location string outputs `52.123123,`

**Comment by jwkerley** [comment]
Yep, that's a bug. I've submitted a new pull request with a fix for 0. Thanks.

**Comment by maximn** [comment]
Are you sure? Can't see any pull requests

**Comment by jwkerley** [comment]
Try now?

**Comment by maximn** [comment]
yap, let's continue on the PR

**Comment by berinder** [comment]
Since fixes for 0.0, scientific notation and now also culture invariant have been merged this issue can be closed.

**Comment by peterdew** [comment]
@maximn, if you agree, please close this issue, so that contributors can focus on other open issues.
